HW22/DynamicArray.h

```diff
@@ -0,0 +1,136 @@
+#pragma once
+#include <cstdlib>
+
+template<typename T>
+class DynamicArray
+{
+public:
+    DynamicArray() : data(nullptr), size(0), capacity(0) {}
+    DynamicArray(std::size_t size) : size(size), capacity(size) { data = new T[size](); }
+    DynamicArray(const DynamicArray& other) : data(nullptr), size(0), capacity(0) { copyData(other); }
+
+    ~DynamicArray() { delete[] data; }
+
+    DynamicArray& operator=(const DynamicArray& other); 
+
+    T& operator[](std::size_t index) {return data[index];}
+
+    void setSize(std::size_t newSize);
+    std::size_t getSize() const {return size;}
+
+    void clear();
+
+    void push_back(T element);
+
+    void pop_back() { if (size > 0) { --size; } }
```

Comment:
same, too much reserve may be left unused. 
In some sense this function should be opposite to push_back(), that is to decrease reserve once in a while